client/my-sites/plans-features-main/hooks/use-free-hosting-trial-assignment.ts

```diff
@@ -1,14 +1,15 @@
 import { isEnabled } from '@automattic/calypso-config';
 import { useExperiment } from 'calypso/lib/explat';
+import type { PlansIntent } from 'calypso/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans';
 
-export function useFreeHostingTrialAssignment(): {
+export function useFreeHostingTrialAssignment( intent: PlansIntent | undefined ): {
 	isLoadingHostingTrialExperiment: boolean;
 	isAssignedToHostingTrialExperiment: boolean;
 } {
 	const [ isLoadingHostingTrialExperiment, experimentAssignment ] = useExperiment(
-		'wpcom_hosting_business_plan_free_trial',
+		'wpcom_hosting_business_plan_free_trial_v2',
 		{
-			isEligible: ! isEnabled( 'plans/hosting-trial' ),
+			isEligible: intent === 'plans-new-hosted-site' && ! isEnabled( 'plans/hosting-trial' ),
```

Comment:
What conditions would a user have the flag `plans/hosting-trial`?

Reply:
Doesn't happen for users, but it's a tool for devs. Makes answering this question easy :) p1699428324232609-slack-C04H4NY6STW

The other way would be to tell them to self-assign their cohort in abacus, that that also requires a discussion about caching in local storage. So if they're not as familiar with abacus it's easier to tell them about the flag.